src/ui/util/columns/getTemplate.ts

```diff
@@ -0,0 +1,13 @@
+import * as templates from "./templates";
+import type { MetaCol } from "./getCols";
+import type { Player } from "../../../common/types";
+import type { TableConfig } from "../TableConfig";
+
+export default function (p: Player, c: MetaCol, config: TableConfig) {
```

Comment:
What if it's a table showing things besides players? Like teams or something else?

Reply:
Well currently it wouldn't be customizable but eventually there would be a set of columns for teams too. I'm open to ideas thou.

Reply:
Is the team data structured similarly enough to the player data where we could have an interface which describes both?

Reply:
I think there's no good way to do it.

Ideally the templates would be generic functions, where the input extends whatever is required. Like

```
export default function({row, c, config}: {
    row: {
        tid: number,
        region: string,
        name: string,
    }
} & TemplateProps) { 
```

Or maybe better:

```
export default function({row, c, config}: TemplateProps<{
    tid: number,
    region: string,
    name: string,
}>) { 
```

Then if we specify what the allowed template functions are for a certain table, I think it would be possible to use TypeScript to take a union of all the `row` types and confirm that the row objects in the table actually conform to the required format. The problem is, the row objects are dynamic based on the selected columns. I think that makes it too difficult to use TypeScript for this. So we'll lose the type safety we currently have (at least for tables displaying data not typed as `any` - many player tables are `any`, but not all, and most non-player tables are not). Which means, like I said in my other comment, this is going to be very hard to do without messing something up.

While I'm rambling here, basically my biggest concerns are:

- This stuff is kind of error prone (what I wrote above in this comment)
- Maybe I need to rewrite perGame/per36/total handling to allow them to be individually selected, before doing column customiation
- What about tables that dynamically show/hide a column based on some conditions (like Player Stats showing All Seasons, or League Stats where past seasons don't have some info) - would need some way to specify that a column, even when selected, may not be possible to display.